### PR TITLE
lab(ch02s09): Developing Cloud-native Microservices with Quarkus

### PR DESCRIPTION
**Pull request by shatakshi06**

### CODE REVIEW

File: develop-lab/solutions/microservice-speaker/src/main/java/org/acme/conference/speaker/SpeakerService.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class SpeakerService {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private SpeakerIdGenerator generator = new SpeakerIdGenerator();
+
+    public Speaker create (Speaker speaker) {
+        //speaker.uuid=generator.generate();
```
Comment (ebramirez) [inline develop-lab/solutions/microservice-speaker/src/main/java/org/acme/conference/speaker/SpeakerService.java:26]: If this generator is not going to be used, this commented line needs to go.

File: develop-lab/solutions/microservice-speaker/src/main/java/org/acme/conference/speaker/SpeakerService.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    public void delete (Speaker speaker) {
+        speaker.delete();
+    }
+
+    public Optional<Speaker> getByUuid(String uuid) {
+        return Speaker.find("uuid", uuid).firstResultOptional();
+    }
+
+    private static class SpeakerIdGenerator {
```
Comment (ebramirez) [inline develop-lab/solutions/microservice-speaker/src/main/java/org/acme/conference/speaker/SpeakerService.java:48]: If this is not used, remove the inner class.

File: develop-lab/solutions/microservice-speaker/src/main/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+# key = value
+%dev.quarkus.http.port=8082
+
+quarkus.datasource.db-kind=h2
+quarkus.datasource.username=admin
+quarkus.datasource.jdbc.url=jdbc:h2:mem:default
+
+quarkus.hibernate-orm.database.generation=drop-and-create
+quarkus.hibernate-orm.sql-load-script=META-INF/import.sql
+quarkus.http.cors=true
```
Comment (ebramirez) [inline develop-lab/solutions/microservice-speaker/src/main/resources/application.properties:11]: This line is duplicated.

File: develop-lab/solutions/microservice-speaker/src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+quarkus.swagger-ui.always-include=true
+quarkus.http.cors=true
```
Comment (ebramirez) [inline develop-lab/solutions/microservice-speaker/src/main/resources/application.properties:14]: If these lines are not strictly necessary, I would remove them.

**Review by ebramirez — CHANGES_REQUESTED** [review]
Good job @shatakshi06!

Just a couple of suggestions to remove, as we usually not push anything that it is not exactly need.

**Comment by shatakshi06** [comment]
> Good job @shatakshi06!
> 
> Just a couple of suggestions to remove, as we usually not push anything that it is not exactly need.

Thank you @ebramirez :)

I will definitely remove all the code which is not needed.

File: develop-lab/solutions/microservice-speaker/.dockerignore [hunk 5]
```diff
@@ -0,0 +1,5 @@
+*
+!target/*-runner
```
[the remark this answers is not in the document]
Comment (shatakshi06) [reply develop-lab/solutions/microservice-speaker/.dockerignore:2]: It is mentioned in the sheet to include "Native Build and Containers".

So I thought to make it a part of the app repo.

File: develop-lab/solutions/microservice-speaker/src/main/java/org/acme/conference/speaker/Speaker.java [hunk 6]
```diff
@@ -41,15 +39,60 @@ public Speaker(String nameFirst, String nameLast, String organization,
...
+  }
+  public void setPicture(String picture) {
+    this.picture = picture;
+  }
+
+  public String getTwitterHandle() {
+    return twitterHandle;
+  }
+  public void setTwitterHandle(String twitterHandle) {
+    this.twitterHandle = twitterHandle;
```
Comment (ebramirez) [inline develop-lab/solutions/microservice-speaker/src/main/java/org/acme/conference/speaker/Speaker.java:90]: With Panache getters and setters are not actually needed, just set the entity's properties public.

You can see this doc for reference: https://quarkus.io/guides/hibernate-orm-panache#how-and-why-we-simplify-hibernate-orm-mappings

**Review by ebramirez — APPROVED** [review]
Just a minor comment, your choice to apply it or not.

File: develop-lab/solutions/microservice-speaker/src/main/java/org/acme/conference/speaker/Speaker.java (hunk 6 above)
Comment (shatakshi06) [reply develop-lab/solutions/microservice-speaker/src/main/java/org/acme/conference/speaker/Speaker.java:90]: Noted!!

File: develop-review/solutions/microservice-speaker/src/main/resources/application.properties [hunk 7]
```diff
@@ -0,0 +1,5 @@
+quarkus.datasource.db-kind = postgresql
+%dev.quarkus.datasource.jdbc.url = jdbc:postgresql://localhost:5432/conference
```
Comment (jramcast) [inline develop-review/solutions/microservice-speaker/src/main/resources/application.properties:2]: The db name is different from the book
```suggestion
%dev.quarkus.datasource.jdbc.url = jdbc:postgresql://localhost:5432/testing
```

**Review by jramcast — CHANGES_REQUESTED** [review]

File: develop-review/solutions/microservice-speaker/src/main/resources/application.properties (hunk 7 above)
Comment (psolarvi) [reply develop-review/solutions/microservice-speaker/src/main/resources/application.properties:2]: oh.. good catch! I renamed the DB when I was creating a generic podman-compose template.

**Review by jramcast — APPROVED** [review]

**Pull request merged**